ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,132 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.equal('dummy');
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should not map model into layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.be.empty;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input').innerText).to.be.empty;
+    });
+
+    test('Should not display the layout when href changes and it is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html"></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+
+      layout.href = '';
+      await flush();
+
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
```

Comment:
This spy is redundant, since we area explicitly calling that method in the following line.

Reply:
Indeed!